plugin/secrets_service_account_key.go

```diff
@@ -195,6 +201,22 @@ func (b *backend) getSecretKey(ctx context.Context, s logical.Storage, rs *RoleS
 		return logical.ErrorResponse(err.Error()), nil
 	}
 
+	if ttl == 0 {
+		ttl = int(cfg.TTL.Seconds())
+	}
+	if ttl == 0 {
+		ttl = int(b.System().DefaultLeaseTTL().Seconds())
+	}
+
+	maxTTL := int(cfg.MaxTTL.Seconds())
+	if maxTTL <= 0 {
+		maxTTL = int(b.System().MaxLeaseTTL().Seconds())
+	}
+
+	if ttl > maxTTL {
+		ttl = maxTTL
+	}
```

Comment:
should we add a warning to the response?

Reply:
I thought about it, but none of the other plugins push out a warning, and the response itself includes the computed TTL.

Reply:
👍 i'm fine either way, just thought it was nicer than ignoring the TTL given

Reply:
I verified Jim's comment below. Turns out we don't need any of this :smile: